src/ucp/tag/rndv.c

```diff
@@ -293,10 +405,30 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_rndv_get_zcopy, (self),
     state = rndv_req->send.state.dt;
     ucp_dt_iov_copy_uct(iov, &iovcnt, max_iovcnt, &state, rndv_req->send.buffer,
                         ucp_dt_make_contig(1), length);
-    status = uct_ep_get_zcopy(rndv_req->send.ep->uct_eps[rndv_req->send.lane],
+
+    if (rndv_req->send.rndv_get.mrail &&
+        ((ucp_request_is_empty_rail(&rndv_req->send.state.dt,
+                                    rndv_req->send.rndv_get.mrail->rail_idx)) ||
+         (rndv_req->send.rndv_get.mrail->rail_idx >= UCP_MAX_RAILS))) {
+        rndv_req->send.rndv_get.mrail->rail_idx = 0;
+    }
+
+    if (!rndv_req->send.rndv_get.mrail) {
```

Comment:
need to unit single/multi lane logic..

Reply:
quite problematic because there is common logic for rndv-get & tag offloading, side effects may be unpredictable